index.bs

```diff
@@ -3296,6 +3300,270 @@ failed</dfn> steps given |context| and |navigation status|:
 
 </div>
 
+## The network Module ## {#module-network}
+
+The <dfn export for=modules>network</dfn> module contains commands and events
+relating to network requests.
+
+### Definition ### {#module-network-definition}
+
+<pre class="cddl remote-cddl">
+
+NetworkCommand = (
+)
+
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+NetworkResult = (
+)
+
+NetworkEvent = (
+    NetworkBeforeSendRequest //
+    NetworkResponseStarted //
+    NetworkResponseCompleted //
+    NetworkError
+)
+
+</pre>
+
+### Types ### {#module-network-types}
+
+#### The network.Cookie type #### {#type-network-Cookie}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkCookie = {
+    name: text,
+    ? value: text,
+    ? binaryValue: [ uint ]
+    domain: text,
+    path: text,
+    expires: uint,
+    size: unit,
+    httpOnly: boolean,
+    secure: boolean,
+    session: boolean,
+    sameSite: "strict" / "lax" / "none",
+};
+</pre>
+
+The <code>NetworkCookie</code> type represents a cookie.
+
+If the cookie value can be represented as a UTF-8 encoded string, the
+<code>value</code> field will be present. Otherwise the <code>binaryValue</code>
+field will be present and consist of an array of integers representing the bytes
+of the cookie value.
+
+#### The network.FetchTimingInfo type #### {#type-network-FetchTimingInfo}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkFetchTimingInfo = {
```

Comment:
This is in parts similar to https://w3c.github.io/navigation-timing/#the-performancetiming-interface, is the intention that the definitions be taken from that spec?

Reply:
Good question! So in my ideal world this would be exactly like performance timing, since that will be easy to specify and implement. But I think performance timing is a bit different from what's in CDP (https://chromedevtools.github.io/devtools-protocol/tot/Network/#type-ResourceTiming) so some input on what's actually required here would be very helpful. @janodvarko might also have some input on what's required here e.g. for HAR export.

Reply:
Performance timing spec is missing "requestEnd" attribute that could be used to calculate time needed to send a request: `requestEnd` - `requestStart` == `wait` in HAR spec. (this might be especially useful for requests sending POST data).

CDP has both `sendStart` and `sendEnd`, so from this perspective CDP timing is closer to HAR (i.e. it has all HAR needs)

HAR also needs `onLoad` and `domContentLoaded` timings. Those are defined by "Navigation Timing interface" https://w3c.github.io/navigation-timing/#dom-performancenavigationtiming

Does CDP has an interface for "NavigationTiming"?

HAR spec request timings: http://www.softwareishard.com/blog/har-12-spec/#timings
HAR spec page timings: http://www.softwareishard.com/blog/har-12-spec/#pageTimings

Reply:
CDP has some events in the Page domain that allow getting timestamps for load-related events e.g. https://chromedevtools.github.io/devtools-protocol/tot/Page/#event-domContentEventFired . I think we could add timestamps to similar events in `browsingContext` for that part of the use case.

Reply:
I just filed https://github.com/w3c/webdriver-bidi/issues/271 to add timestamps to `browsingContext` related load events.